skymp5-server/cpp/server_guest_lib/script_classes/PapyrusFaction.cpp

```diff
@@ -0,0 +1,34 @@
+#include "PapyrusFaction.h"
+
+VarValue PapyrusFaction::GetReaction(VarValue self,
+                                     const std::vector<VarValue>& arguments)
+{
+  if (arguments.empty()) {
+    spdlog::error("Faction.GetReaction - at least one argument expected");
+    return VarValue(-1);
+  }
+
+  if (arguments[0].GetType() != VarValue::kType_Object) {
+    spdlog::error("Faction.GetReaction - wrong object type passed");
+    return VarValue(-1);
+  }
+
+  espm::LookupResult faction = GetRecordPtr(self);
+  espm::LookupResult otherFaction = GetRecordPtr(arguments[0]);
+
+  WorldState* worldState = compatibilityPolicy->GetWorldState();
+
+  espm::FACT::Data otherFactionData =
+    espm::Convert<espm::FACT>(otherFaction.rec)
+      ->GetData(worldState->GetEspmCache());
+
+  int reaction = -1;
+  for (const auto& interfactionRelation :
+       otherFactionData.interfactionRelations) {
```

Comment:
this is not our reaction. this is _their_ reaction to us

I guess this is a bug